src/liballoc/tests/btree/map.rs

```diff
@@ -142,6 +173,82 @@ fn test_iter_rev() {
     test(size, map.into_iter().rev());
 }
 
+fn do_test_iter_mut_mutation<T>(size: usize)
+where
+    T: Copy + Debug + Ord + TryFrom<usize>,
+    <T as std::convert::TryFrom<usize>>::Error: std::fmt::Debug,
+{
+    let zero = T::try_from(0).unwrap();
+    let mut map: BTreeMap<T, T> = (0..size).map(|i| (T::try_from(i).unwrap(), zero)).collect();
+
+    // Iterate forwards, trying to mutate to unique values
+    for (i, (k, v)) in map.iter_mut().enumerate() {
```

Comment:
Should we make some effort to also check that the loop actually runs `size` many iterations?

Reply:
I actually had that check at first, but the test cases above it already check that (for each type of iterator and in both directions). But I wish there were a way to express such dependencies in unit tests. Maybe that is just repeating it.

Reply:
Ah, true. Seems odd to make this two separate tests but whatever. I'll leave it up to you if you want to test this also here.

Reply:
I've put it back with a comment, it's short and actually more distracting when it's missing.

Reply:
Put what back?

Reply:
The assert that the iteration does `size` iterations.

Reply:
Oh, you are just testing the `count` function. I was looking for code that actually counts the steps. Yeah fair enough. ;)